homeassistant/components/evohome.py

```diff
@@ -0,0 +1,667 @@
+        self._icon = "mdi:thermostat"
+
+        # Set the entity's supported features
+        self._supported_features = SUPPORT_OPERATION_MODE | SUPPORT_AWAY_MODE
+
+        # Set the entity's operation list - hard-coded for a particular order,
+        # instead of using self._config['allowedSystemModes']
+        self._op_list = [
+            EVO_RESET,
+            EVO_AUTO,
+            EVO_AUTOECO,
+            EVO_AWAY,
+            EVO_DAYOFF,
+            EVO_CUSTOM,
+            EVO_HEATOFF
+        ]
+
+        # Create timers, etc. - they're maintained in update(), or schedule()
+        self._status = {}
+        self._timers = domain_data['timers']
+
+        self._timers['statusUpdated'] = datetime.min
+        domain_data['schedules'] = {}
+
+        # Set the entity's (initial) behaviour
+        self._available = False  # will be True after first update()
+        self._should_poll = True
+
+        # Create a listener for (internal) update packets...
+        hass.helpers.dispatcher.async_dispatcher_connect(
+            DISPATCHER_EVOHOME,
+            self._connect
+        )  # for: def async_dispatcher_connect(signal, target)
+
+    def _handle_requests_exceptions(self, err_type, err):
+
+        domain_data = self.hass.data[DATA_EVOHOME]
+
+# evohomeclient1 (<=0.2.7) does not have a requests exceptions handler, but
+# they will manifest as:
+#     File ".../evohomeclient/__init__.py", line 33, in _populate_full_data
+#       userId = self.user_data['userInfo']['userID']
+#   TypeError: list indices must be integers or slices, not str
+
+# ...we can extract the response, which may be like this:
+# {
+#   'code':    'TooManyRequests',
+#   'message': 'Request count limitation exceeded, please try again later.'
+# }
+
+        if err_type == "TooManyRequests":  # not actually from requests library
+            # v1 api limit has been exceeded
+            old_scan_interval = domain_data['params'][CONF_SCAN_INTERVAL]
+            new_scan_interval = min(old_scan_interval * 2, 300)
+            domain_data['params'][CONF_SCAN_INTERVAL] = new_scan_interval
+
+            _LOGGER.warning(
+                "v1 API rate limit has been exceeded, suspending polling "
+                "for %s seconds, & increasing '%s' from %s to %s seconds.",
+                new_scan_interval * 3,
+                CONF_SCAN_INTERVAL,
+                old_scan_interval,
+                new_scan_interval
+            )
+
+            domain_data['timers']['statusUpdated'] = datetime.now() + \
+                timedelta(seconds=new_scan_interval * 3)
+
+# evohomeclient2 (>=0.2.7) now exposes requests exceptions, e.g.:
+# - "Connection reset by peer"
+# - "Max retries exceeded with url", caused by "Connection timed out"
+#       elif err_type == "ConnectionError":  # seems common with evohome
+#           pass
+
+# evohomeclient2 (>=0.2.7) now exposes requests exceptions, e.g.:
+# - "400 Client Error: Bad Request for url" (e.g. Bad credentials)
```

Comment:
No need to describe standard HTTP error codes

Reply:
What I meant to do here was to document what these HTTP error codes meant in the _context of this application_.

I will change rewrite these comments, putting the emphasis on the text inside the brackets

Reply:
What I meant to do here was to document what these HTTP error codes meant in the _context of this application_.

I will change rewrite these comments, putting the emphasis on the text inside the brackets

Reply:
This section has been completely re-written, to address the above, and remove 0-indent comments.